uint8 mode 
---
bool is_success
